docs/sphinx/examples/python/tutorials/quantum_teleporation.ipynb

```diff
@@ -0,0 +1,402 @@
+    "\n",
+    "state_of_q2 = partial_trace(state_vector = np.array(snapshot4), trace_systems=[0,1], dimensions=[2,2,2]) #trace out qubits 0 and 1 leaving us with qubit 2\n",
+    "\n",
+    "np.allclose(state_of_q0, state_of_q2)\n"
+   ]
+  },
+  {
+   "attachments": {},
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "We started with a 3 qubit quantum circuit with the 0<sup>th</sup> qubit in $\\ket{\\psi}$ and the 2<sup>nd</sup> qubit being a portion of the maximally entangled bell state $\\ket{\\beta_{00}}$.\n",
+    "\n",
+    "If we look closely at the code cell above, the output state on the 2<sup>nd</sup> qubit is the same as the random input state, $\\ket{\\psi}$, on the 0<sup>th</sup> qubit. We have **teleported** a quantum state from one qubit to another. There is nothing to restrict how close qubits 0 and 2 have to be since they have no entangling operations between them. They can be placed in different labs or infinitely far apart. \n",
+    "\n",
+    "Moreover, we still obey the no-cloning theorem since the initial state $\\ket{\\psi}$ no longer exists on the 0<sup>th</sup> qubit and hence we only have 1 copy of it as shown below \n"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 18,
+   "metadata": {},
+   "outputs": [
+    {
+     "data": {
+      "text/plain": [
+       "False"
+      ]
+     },
+     "execution_count": 18,
+     "metadata": {},
+     "output_type": "execute_result"
+    }
+   ],
+   "source": [
+    "#Using the final snapshot, we show that psi has been teleported and only 1 copy of psi is preserved\n",
+    "\n",
+    "state_of_q0 = partial_trace(state_vector = np.array(snapshot4), trace_systems=[1,2], dimensions=[2,2,2]) #trace out qubits 1 and 2 leaving us with qubit 0\n",
+    "\n",
+    "state_of_q2 = partial_trace(state_vector = np.array(snapshot4), trace_systems=[0,1], dimensions=[2,2,2]) #trace out qubits 0 and 1 leaving us with qubit 2\n",
+    "\n",
+    "np.allclose(state_of_q0, state_of_q2)\n"
+   ]
+  },
+  {
+   "attachments": {},
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "None of this violates the universal limit of information travelling faster than the speed of light. Notice how we have conditioned operations meaning we have to communicate classical information via classical channels which adhere to this limit."
+   ]
+  },
+  {
+   "attachments": {},
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Teleportation explained\n",
+    "\n",
+    "We have stepped through the circuit above using code. We will now do the same using mathematics to enhance our understanding. \n",
+    "\n",
+    "\n",
+    "We start by gathering experimentalists in a lab and performing the following\n",
+    "$$ \n",
+    "\\begin{aligned} \n",
+    "\n",
+    "\n",
+    "CX_{01}H_{0}\\ket{00}    &=  CX_{01}\\tfrac{1}{\\sqrt{2}}(\\ket{0}+\\ket{1})\\ket{0} \\\\\n",
```

Comment:
This isn't rendering properly on github -- there's a chance it will be fine in the published docs, but we should double check before any merging